### ISSUE TITLE
Benchmark ABCI methods

### ISSUE THREAD

**Issue by evan-forbes** [celestiaorg/celestia-app#2009]
We're adding the ability to profile with https://github.com/celestiaorg/celestia-core/issues/1004, but it might also be useful to have traditional go benchmarks for each of the celestia-specific ABCI methods (`CheckTx`, `PrepareProposal`, and `ProcessProposal`)

**Comment by starlabman** [comment]
import subprocess

def remplir_bloc():
    commande = "txsim" # Remplacez "txsim" par la commande que vous souhaitez exécuter
    
    try:
        # Exécuter la commande et capturer la sortie
        resultat = subprocess.run(commande, capture_output=True, text=True, shell=True)
        
        if resultat.returncode == 0:
            # La commande s'est exécutée avec succès
            sortie = resultat.stdout.strip() # Obtenez la sortie de la commande
            # Traitez la sortie comme vous le souhaitez
            
            # Remplir le bloc avec la sortie de la commande
            bloc = sortie  # Remplacez "sortie" par la valeur que vous souhaitez mettre dans le bloc
            
            return bloc
        else:
            # La commande a retourné une erreur
            erreur = resultat.stderr.strip() # Obtenez l'erreur de la commande
            # Traitez l'erreur comme vous le souhaitez
            
            return None
    except subprocess.CalledProcessError as e:
        # Une erreur s'est produite lors de l'exécution de la commande
        print("Erreur lors de l'exécution de la commande :", e)
        return None

# Appeler la fonction pour remplir le bloc
bloc_rempli = remplir_bloc()

if bloc_rempli is not None:
    print("Le bloc a été rempli avec succès :", bloc_rempli)
else:
    print("Une erreur s'est produite lors du remplissage du bloc.")


J'espère que cela vous aidera à remplir le bloc en utilisant la commande "txsim" avec Python! Si vous avez d'autres questions, n'hésitez pas à demander. https;//github.com/starlabman

**Comment by starlabman** [comment]
import subprocess

def remplir_bloc():
    commande = "txsim" # Remplacez "txsim" par la commande que vous souhaitez exécuter
    
    try:
        # Exécuter la commande et capturer la sortie
        resultat = subprocess.run(commande, capture_output=True, text=True, shell=True)
        
        if resultat.returncode == 0:
            # La commande s'est exécutée avec succès
            sortie = resultat.stdout.strip() # Obtenez la sortie de la commande
            # Traitez la sortie comme vous le souhaitez
            
            # Remplir le bloc avec la sortie de la commande
            bloc = sortie  # Remplacez "sortie" par la valeur que vous souhaitez mettre dans le bloc
            
            return bloc
        else:
            # La commande a retourné une erreur
            erreur = resultat.stderr.strip() # Obtenez l'erreur de la commande
            # Traitez l'erreur comme vous le souhaitez
            
            return None
    except subprocess.CalledProcessError as e:
        # Une erreur s'est produite lors de l'exécution de la commande
        print("Erreur lors de l'exécution de la commande :", e)
        return None

    # Appeler la fonction pour remplir le bloc
bloc_rempli = remplir_bloc()

if bloc_rempli is not None:
    print("Le bloc a été rempli avec succès :", bloc_rempli)
else:
    print("Une erreur s'est produite lors du remplissage du bloc.")




#J'espère que cela vous aidera à remplir le bloc en utilisant la commande "txsim" avec Python! Si vous avez d'autres questions, #n'hésitez pas à demander. https;//github.com/starlabman

**Comment by rawbytedev** [comment]
hello @evan-forbes 
Does traditional benchmarking refer to using b.N loop instead of reporting manually? Could you clarify it?

**Comment by rootulp** [comment]
Yes please use [Go benchmarks](https://pkg.go.dev/testing#hdr-Benchmarks) for this issue.

**Comment by rawbytedev** [comment]
Thanks
